### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry/StructureSheaf): add two useful lemmas

### PR DESCRIPTION
**Pull request by jjaassoonn**
Let $f : R \to S$ be a ring map

1.  The preimage of some basic open set $D(x)$ under $\mathrm{Spec}f$ is $D(f(x))$ (proof is `rfl`)
2.  the ring map $\Gamma(\mathcal{O} _ {\mathrm{Spec}R}, D(x)) \to \Gamma(\mathcal{O}_{\mathrm{Spec} S}, D(f(x))$ is the map between localizations `R_x` and `S_{f(x)}.`

These are lemmas are found useful in #12371 by Andrew.

Co-authored-by: Andrew Yang <the.erd.one@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean [hunk 1]
```diff
@@ -885,6 +885,10 @@ theorem isCompact_basicOpen (f : R) : IsCompact (basicOpen f : Set (PrimeSpectru
   exact isCompact_range (map_continuous _)
 #align prime_spectrum.is_compact_basic_open PrimeSpectrum.isCompact_basicOpen
 
+lemma comap_basicOpen {R S} [CommSemiring R] [CommSemiring S] (f : R →+* S) (x : R) :
```
Comment (erdOne) [inline Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:888]:
```suggestion
lemma comap_basicOpen (f : R →+* S) (x : R) :
```

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors d+

**Pull request closed without merging**